modules/core/include/opencv2/core/ocl.hpp

```diff
@@ -331,8 +331,12 @@ class CV_EXPORTS Queue
     void* ptr() const;
     static Queue& getDefault();
 
+    /// @brief Returns OpenCL command queue with enable profiling mode support
+    const Queue& getProfilingQueue() const;
+
+    struct Impl; friend struct Impl;
```

Comment:
is this part really necessary? looks like we break encapsulation, but usually we find a way to keep it

Reply:
This is still inaccessible via public API. Alternate way is to extend public API with new "private" methods that I want to avoid via this construction.